system/smp.c

```diff
@@ -515,6 +515,38 @@ static bool start_cpu(int cpu_num)
     return true;
 }
 
+bool check_x2apic(void)
```

Comment:
+static, given that this is a function with a single caller, which hasn't (yet) been exported to smp.h.

Reply:
Yes, but this function is not executed (because the asm code in startup.S has already been executed on the BSP, so x2apic is already disabled when the code goes here). I'm doing some tests to find the right way to call check_x2apic() externally from startup.S, but it's more complex than expected.